### DISCUSSION TITLE
Queueing a callback to be invoked just once

### DISCUSSION THREAD

**Discussion by cole-miller** [libuv/libuv#4084] [Q&A] [answered]
What is the recommended approach for registering a callback to be invoked by an event loop just once, as opposed to every loop iteration? The first thing I thought of was to use a uv_idle_t handle with a callback that calls uv_idle_stop before doing whatever its real job is -- is that safe (i.e. is uv_idle_stop reentrant)? Is there a better way?

**Comment by bnoordhuis** [comment, marked as answer]
Yes, that works. A zero timeout timer would work too.

**Reply by cole-miller** [reply]
Oh, the timer is nicer, thanks!
